Mnt 16 feature add loading component

Jade-Dragon88 opened the pull request:
1. Добавил другие виды нотификации в notification.
2. Сделал notification немутабельным.

Jade-Dragon88 committed:
Добавил loading в таблицу учеников, удалил error

Jade-Dragon88 committed:
Добавил в MentiList нотификацию, запроксировал стандартную ant-функцию Notification в файле helpers\notification.ts

Jade-Dragon88 committed:
1. Добавил другие виды нотификации в notification
2. Сделал notification не мутабельным

ivnvMkhl commented on src/helpers/notification/notification.ts:
```diff
@@ -0,0 +1,31 @@
+  success: (config: NotificationConfig) => {
+    antdNotification.success(makeNotificationConfig(config));
+  },
+};
```
Давай еще к объекту нотификации добавим `as const ` чтобы его невозможно было мутировать где то во внешней среде

ivnvMkhl commented on src/helpers/notification/notification.ts:
```diff
@@ -0,0 +1,31 @@
+type NotificationConfig = ArgsProps;
+
+const DEFAULT_DURATION = 15;
+const DEFAULT_PLACEMENT = 'bottomRight';
```
давай константы которые относятся к нотификации вынесем в отдельный файл `notification.constants.ts`

ivnvMkhl commented on src/helpers/notification/notification.ts:
```diff
@@ -0,0 +1,31 @@
+import { notification as antdNotification } from 'antd';
+import type { ArgsProps } from 'antd/es/notification';
+
+type NotificationConfig = ArgsProps;
```
типы вынесем в файл `notification.interfaces.ts`

ivnvMkhl commented on src/helpers/notification/notification.ts:
```diff
@@ -0,0 +1,31 @@
+const DEFAULT_DURATION = 15;
+const DEFAULT_PLACEMENT = 'bottomRight';
+
+const defaultConfig: Partial<NotificationConfig> = {
```
Это тоже константа, во первых по неймингу это набор констант поэтому `DefaultConfig` (перечитай пункт код гайда 1.4 ) и добавляем `as const` и выносим в файл констант
![image](https://github.com/ivnvMkhl/mentoring-ui/assets/99988140/640b3bfd-6726-4a2c-9a75-588a00047be6)

ivnvMkhl commented on src/components/pages/MentiList/MentiList.tsx:
```diff
@@ -58,7 +61,7 @@ const MentiList: FC = () => {
           { title: 'Телефон', dataIndex: 'Phone', key: 'Phone' },
         ]}
```
Давай этот список с конфигом таблицы вынесем в константы так же в файл `MentiList.constnts.ts`

ivnvMkhl commented on src/components/pages/MentiList/MentiList.tsx:
```diff
@@ -21,27 +22,29 @@ type Student = {
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     fetch('https://imkhl-mentoring-1.glitch.me/menti')
```
Url для запроса так же поидее надо выносить в константы но мы это пока откладываем нам нужно будет сделать api  сервис в котором мы сделаем функции для запросов разных сущностей. Пока тут можно оставить TODO о том что пока запрос написан так до создания api сервиса

ivnvMkhl commented on src/components/pages/MentiList/MentiList.tsx:
```diff
@@ -5,6 +5,7 @@ import { Table } from '../../primitives/Table/Table';
 import { PageWrapper } from '../../complex/PageWrapper/PageWrapper';
+import { notification } from '../../../helpers/notification/notification';
 
 type Student = {
```
Давай по неймингу будем соблюдать согласованность с бекендом, эта сущность называется `Menti` и этот тип нужно вынести в общие ресурсы для проекта `/ src / interfaces / menti.interfaces.ts`

ivnvMkhl commented on src/components/pages/MentiList/MentiList.tsx:
```diff
@@ -21,27 +22,29 @@ type Student = {
 };
 
 const MentiList: FC = () => {
-  const [students, setStudents] = useState<Student[]>();
```
тут тоже `students` -> `mentiList`

Jade-Dragon88 committed:
1. Добавил "as const" к объекту нотификации ( файл notification.ts )
2. Константы DEFAULT_DURATION и DEFAULT_PLACEMENT, а так же type NotificationConfig вынес в отдельный файл notification.constants.ts
3. DefaultConfig вынес в отдельный  файл констант notification.constants.ts
4. Список с конфигом таблицы colums вынес в отдельный файл MentiList.constants.ts
5. type Students переименовал в type Menti и вынес в отдельный файл menti.interfaces.ts
6. Переменную students переименовал в mentiList ( файл MentiList.tsx )

ivnvMkhl commented:
@Jade-Dragon88  у тебя куда то 4 файла потерялись, видишь сборка тестовая упала? 
![image](https://github.com/ivnvMkhl/mentoring-ui/assets/99988140/95695624-6a73-405c-be16-0c21c522c998)

Jade-Dragon88 committed:
Добавил в отслеживание недостающие файлы
